src/main/java/core/basesyntax/model/FruitRecord.java

```diff
@@ -0,0 +1,65 @@
+package core.basesyntax.model;
+
+public class FruitRecord {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitRecord(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+
+            this.code = code;
+        }
+
+        public String getCode() {
+
+            return code;
+        }
+
+        public static Operation fromCode(String code) {
+            for (Operation operation : values()) {
+                if (operation.getCode().equals(code)) {
+                    return operation;
+                }
+            }
+            throw new IllegalArgumentException("Unknown operation: " + code);
+        }
+    }
```

Comment:
The `Operation` enum is public and nested within `FruitRecord`. If it is intended to be used outside of `FruitRecord`, it would be better to place it in its own file to follow the encapsulation principle and to make the code structure clearer, unless there's a specific reason it needs to be nested.